internal/v2/cert_verifier/cert_verifier.go

```diff
@@ -0,0 +1,62 @@
+// Certificate Verification that offloads operations to S2Av2.
+package certverifier
+
+import (
+	"crypto/x509"
+	"fmt"
+	"time"
+)
+
+// VerifyClientCertificateChain returns any errors found during certificate verification.
```

Comment:
Can we add a slightly-more descriptive comment? For example:

// VerifyClientCertificateChain verifies a client's certificate chain against the roots in the provided pool and checks that the common name in the client's leaf certificate matches the expected common name.

Similarly for the server cert API.